solidity/contracts/TokenStakingEscrow.sol

```diff
@@ -247,6 +269,19 @@ contract TokenStakingEscrow is Ownable {
         emit DepositWithdrawn(operator, grantee, amount);
     }
 
+    function withdrawRevoked(
+        Deposit memory deposit,
+        address operator,
+        address grantManager
+    ) internal {
+        uint256 amount = deposit.amount - deposit.withdrawn;
```

Comment:
Are we certain that the amount calculated this way is always equal to the revoked amount?

Reply:
It's not equal to the revoked amount and your question motivated me to improve docs even further in https://github.com/keep-network/keep-core/pull/1872/commits/f7ba39562623f4de09f94e307d53d2bfe52b97c6. 

We can't calculate the exact amount for a revoked grant that should be withdrawable from escrow. There are different staking policies, we are going to introduce stake top-ups, part of the tokens could be withdrawn before... It's just too hard if possible at all. We do a simplification:
- grant not revoked - let grantee withdraw based on the unlocking schedule
- grant revoked but fully unlocked - let grantee withdraw everything
- grant revoked and not fully unlocked - do not let grantee withdraw more than already withdrawn; grant manager takes the rest and they may split the tokens out of band.